### PR TITLE
Add design restraints

### PR DESCRIPTION
**Pull request by haslinghuis**

### CODE REVIEW

File: docs/development/manufacturer/manufacturer-design-guidelines.md [hunk 1]
```diff
@@ -294,6 +294,10 @@ For details of the use of these LEDs, please see the [FC LEDs](/docs/development
 
 ## 3.2 Resource Selection Considerations
 
+- Betaflight does not support sharing devices on the SPI bus which is blocking excution and results in bad performance. Mainly sharing MAX7456 and blackbox generates support issues.
+- Optimal motors assignment would use the same channel requiring only one DMA channel used for dshot telemetry with bitbang enabled.
+- Do not use advanced timers 1 or 8 for anything other then motors.
```
Comment (nerdCopter) [inline docs/development/manufacturer/manufacturer-design-guidelines.md:299]:
```suggestion
- Do not use advanced timers 1 or 8 for anything other than motors.
```

**Review by nerdCopter — DISMISSED** [review]

File: docs/development/manufacturer/manufacturer-design-guidelines.md [hunk 2]
```diff
@@ -294,6 +294,10 @@ For details of the use of these LEDs, please see the [FC LEDs](/docs/development
 
 ## 3.2 Resource Selection Considerations
 
+- Betaflight does not support sharing devices on the SPI bus which is blocking excution and results in bad performance. Mainly sharing MAX7456 and blackbox generates support issues.
+- Optimal motor assignment would use the same channel each four motors requiring only one DMA channel used for dshot telemetry with bitbang enabled.
```
Comment (KarateBrot) [inline docs/development/manufacturer/manufacturer-design-guidelines.md:298]: I don't really understand what this means. Can we split this up into multiple sentences?

Comment (haslinghuis) [reply docs/development/manufacturer/manufacturer-design-guidelines.md]: Agree - a bit too cryptic. Updated.

**Review by nerdCopter — APPROVED** [review]

File: docs/development/manufacturer/manufacturer-design-guidelines.md [hunk 3]
```diff
@@ -294,6 +294,23 @@ For details of the use of these LEDs, please see the [FC LEDs](/docs/development
...
+- 8 motors spread across more than 2 GPIO ports is BAD.
+
+Similarly, it is optimal to use two 4-channel timers for 8 motors for when BITBANG is disabled.
+
+There is also a choice between using advanced timers or not, TIM1/TIM8 are advanced and get used by DSHOT BITBANG.
+
+It may be optimal to use TIM1 + TIM8 for all motors so that the other timers are always free.
+Or it may be optimal to use timers other than TIM1/TIM8 for motors so that TIM1/TIM8 are free for other other uses when DSHOT BITBANG is NOT used.
+
+Note: TIM1 has inter-peripheral connectivity that other timers do not have.
```
Comment (KarateBrot) [inline docs/development/manufacturer/manufacturer-design-guidelines.md:312]:
```suggestion
::: note

TIM1 has inter-peripheral connectivity that other timers do not have.

:::
```

File: docs/development/manufacturer/manufacturer-design-guidelines.md [hunk 4]
```diff
@@ -294,6 +294,23 @@ For details of the use of these LEDs, please see the [FC LEDs](/docs/development
 
 ## 3.2 Resource Selection Considerations
 
+- Betaflight does not support sharing devices on the SPI bus which is blocking excution and results in bad performance. Mainly sharing MAX7456 and blackbox generates support issues.
```
Comment (KarateBrot) [inline docs/development/manufacturer/manufacturer-design-guidelines.md:297]:
```suggestion
::: warning

Betaflight does not support sharing devices on the SPI bus which is blocking excution and results in bad performance. Mainly sharing MAX7456 and blackbox generates support issues.

:::
```

**Review by KarateBrot — CHANGES_REQUESTED** [review]

**Review by nerdCopter — APPROVED** [review]

**Review by nerdCopter — APPROVED** [review]

**Comment by haslinghuis** [comment]
@KarateBrot change request had spaces after `:::`. Fixed now.

**Comment by nerdCopter** [comment]
i wish this repo would not reset approvals

**Review by ot0tot — APPROVED** [review]

**Pull request merged**